Add ImageNet-R to tfds

hendrycks opened the pull request:
# Add Dataset

* Dataset Name: ImageNet-R
  
## Description

ImageNet-R(endition) contains art, cartoons, deviantart, graffiti, embroidery, graphics, origami, paintings, patterns, plastic objects, plush objects, sculptures, sketches, tattoos, toys, and video game renditions of ImageNet classes.
ImageNet-R has renditions of 200 ImageNet classes resulting in 30,000 images.

This was created by modifying the imagenet_a.py file.

## Checklist
* [ ] Address all TODO's
* [ ] Add alphabetized import to subdirectory's `__init__.py`
* [ ] Run `download_and_prepare` successfully
* [ ] Add [checksums file](https://www.tensorflow.org/datasets/add_dataset#2_run_download_and_prepare_locally)
* [ ] Properly cite in `BibTeX` format
* [ ] Add passing test(s)
* [ ] Add test data
* [ ] If using additional dependencies (e.g. `scipy`), use [lazy_imports](https://www.tensorflow.org/datasets/add_dataset#extra_dependencies) (if applicable)
* [ ] Add data generation script (if applicable)
* [ ] [Lint](https://www.tensorflow.org/datasets/add_dataset#5_check_your_code_style) code

hendrycks commented:
@googlebot I signed it!

josipd commented on tensorflow_datasets/image_classification/imagenet_r_test.py:
```diff
@@ -0,0 +1,34 @@
+      'test': 10,  # Number of fake test examples.
+  }
+
+  DL_EXTRACT_RESULT = 'imagenet-r.tar'
```
Did you forge to add this directory? I guess just copying the Imagenet-A, renaming to imagenet-r.tar and changing the synsets by adding some _suffixes should work.

hendrycks commented:
I replaced imagenet-r.tar with a new tar that just has imagenet_synset_id without the human-readable label (my choice to include the suffix was too nonstandard). Consequently `class_synset.split("_")[0]` won't be necessary, so hopefully the commit is now fine.

josipd commented:
Ah, I see, sorry. Then I guess what remains is to create a copy of for your dataset (same directory structure, but s/imagenet_a/imagenet_r) https://github.com/tensorflow/datasets/tree/master/tensorflow_datasets/testing/test_data/fake_examples/imagenet_a/imagenet-a.tar/imagenet-a so that the tests will pass.

hendrycks commented:
Hopefully it is now addressed.

josipd commented:
I just checked the .tar file and the structure is still unchanged, e.g.`imagenet-r/n04266014_space_shuttle/sculpture_5.jpg` Did you forget to update the link, or should we change the code?

hendrycks commented:
I think the file didn't sync earlier. I re-uploaded the modified version, downloaded it, and checked that now it should be right.

josipd commented:
Works now, thanks!
